operator/controller/src/test/resources/podtemplatespec-test-cases.yaml

```diff
@@ -0,0 +1,481 @@
+              exec:
+                command:
+                  - cat
+                  - /tmp/healthy
+                initialDelaySeconds: 5
+                periodSeconds: 5
+
+  - id: 'Liveness probe empty'
+    original:
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            livenessProbe:
+              httpGet:
+                path: /health/live
+                port: 8080
+                scheme: HTTP
+    spec: { }
+    expected:
+      metadata: { }
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            livenessProbe:
+              httpGet:
+                path: /health/live
+                port: 8080
+                scheme: HTTP
+
+  - id: 'Readiness probe 1'
+    original:
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            livenessProbe:
+              httpGet:
+                path: /health/live
+                port: 8080
+                scheme: HTTP
+    spec:
+      spec:
+        containers:
+          - name: test
+            livenessProbe:
+              exec:
+                command:
+                  - cat
+                  - /tmp/live
+                initialDelaySeconds: 5
+                periodSeconds: 5
+    expected:
+      metadata: { }
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            livenessProbe:
+              exec:
+                command:
+                  - cat
+                  - /tmp/live
+                initialDelaySeconds: 5
+                periodSeconds: 5
+
+  - id: 'Resources empty'
+    original:
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            resources:
+              requests:
+                cpu: 500m
+                memory: 512Mi
+              limits:
+                cpu: '1'
+                memory: 1Gi
+    spec: { }
+    expected:
+      metadata: { }
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            resources:
+              requests:
+                cpu: 500m
+                memory: 512Mi
+              limits:
+                cpu: '1'
+                memory: 1Gi
+
+  - id: 'Resources 1'
+    original:
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            resources:
+              requests:
+                cpu: 500m
+                memory: 512Mi
+              limits:
+                cpu: '1'
+                memory: 1Gi
+    spec:
+      spec:
+        containers:
+          - name: test
+            resources:
+              requests:
+                cpu: 750m
+              limits:
+                memory: 2Gi
+    expected:
+      metadata: { }
+      spec:
+        containers:
+          - name: test
+            image: test:tag
+            resources:
+              requests:
+                cpu: 750m
```

Comment:
Looking at this example, I start to think that the approach used for the probes is the most straightforward to understand and reason about:
if you define a resource request or limit, then it will override the entire object.

We also need to define the semantics for "removing" things.

Reply:
The difference between probes and R&R is that for probes, (I think) you usually define a single kind of probe, and rarely change only some specific setting such as `initialDelaySeconds`. It is also harder to merge than R&L, since they are simple maps.

> We also need to define the semantics for "removing" things.
Can you clarify please?

Reply:
> Can you clarify please?

the practical use case is:
how do I remove the `cpu` limit using `PodTemplateSpec`?

I think I like the most the full override approach used for probes, do you see any issue with that?

Reply:
In the current approach, users can "remove" the cpu limit by setting a different one, I assume it's very rare where users would want to only remove the limit, but I don't have a problem with that.

Reply:
> In the current approach, users can "remove" the cpu limit by setting a different one, I assume it's very rare where users would want to only remove the limit

Setting a different value doesn't remove the limit, I have seen a lot of use cases where people are removing the CPU limit to avoid the applications to be throttled(which is a condition hard to identify in a live cluster).